bridges/snowbridge/pallets/system-v2/src/lib.rs

```diff
@@ -141,40 +136,29 @@ pub mod pallet {
 		/// Sends a command to the Gateway contract to instantiate a new agent contract representing
 		/// `origin`.
 		///
-		/// Fee required: Yes
-		///
-		/// - `origin`: Must be `Location` of a sibling parachain
+		/// - `location`: The location representing the agent
+		/// - `fee`: Ether to pay for the execution cost on Ethereum
 		#[pallet::call_index(1)]
 		#[pallet::weight(T::WeightInfo::create_agent())]
 		pub fn create_agent(
 			origin: OriginFor<T>,
 			location: Box<VersionedLocation>,
 			fee: u128,
 		) -> DispatchResult {
-			let _ = match origin.clone().into() {
-				Ok(RawOrigin::Root) => Ok(Here.into()),
-				_ => T::SiblingOrigin::ensure_origin(origin),
-			}?;
+			T::FrontendOrigin::ensure_origin(origin)?;
 
 			let origin_location: Location =
 				(*location).try_into().map_err(|_| Error::<T>::UnsupportedLocationVersion)?;
 
-			let ethereum_location = T::EthereumLocation::get();
-			// reanchor to Ethereum context
-			let location = origin_location
-				.clone()
-				.reanchored(&ethereum_location, &T::UniversalLocation::get())
-				.map_err(|_| Error::<T>::LocationConversionFailed)?;
-
-			let agent_id = agent_id_of::<T>(&location)?;
+			let agent_id = Self::location_to_agent_id(&origin_location)?;
 
 			// Record the agent id or fail if it has already been created
 			ensure!(!Agents::<T>::contains_key(agent_id), Error::<T>::AgentAlreadyCreated);
 			Agents::<T>::insert(agent_id, ());
 
 			let command = Command::CreateAgent {};
 
-			Self::send(origin_location.clone(), command, fee)?;
+			Self::send(agent_id, command, fee)?;
```

Comment:
A hashed origin location may or may not correspond to the ID of an Agent contract on ethereum.

```suggestion
			Self::send(location_hash, command, fee)?;
```